### DISCUSSION TITLE
Consider running everything in a sandbox prohibiting reads for sensitive files

### DISCUSSION THREAD

**Discussion by mxcl** [pkgxdev/pkgx#605] [Ideas]
macOS does this automatically for some stuff, but not for anything OSS related.

eg. prohibiting reads of `~/.ssh` is an easy security win with few downsides. Tools that genuinely need access can request it in the package yaml and then tea can prompt to allow it on first run.

Ideally I think we'd extend the macOS security feature (no idea what system controls this, you see it if you do a `find ~` for example). However I don’t think that's possible.

**Comment by yozlet** [comment]
The good news is that Mac process sandboxing seems to be highly configurable and can be dynamically invoked for individual processes. Here's a shell tool built around it: https://github.com/lynaghk/sandboxtron

The bad news is that the underlying macOS tool, `sandbox-exec`, has _apparently_ been deprecated for a few years. But much of macOS's internal systems still depend on it, so it's probably not disappearing for a while. More here: https://www.karltarvas.com/2020/10/25/macos-app-sandboxing-via-sandbox-exec.html
